realm/realm-library/src/androidTest/kotlin/io/realm/EmbeddedObjectsTest.kt

```diff
@@ -132,11 +130,79 @@ class EmbeddedObjectsTest {
     }
 
     @Test
-    @Ignore("Placeholder for all tests for DynamicRealm.createEmbeddedObject()")
-    fun dynamicRealm_createEmbeddedObject() {
-        TODO()
+    fun dynamicRealm_createEmbeddedObject() =
+            DynamicRealm.getInstance(realm.configuration).use { realm ->
+                realm.executeTransaction {
+                    val parent = realm.createObject("EmbeddedSimpleParent", "PK_VALUE")
+                    val child = realm.createEmbeddedObject("EmbeddedSimpleChild", parent, "child")
+
+                    val idValue = "ID_VALUE"
+                    child.setString("id", idValue)
+
+                    val childInParent = parent.getObject("child")
+                    assertNotNull(childInParent)
+                    assertEquals(childInParent!!.getString("id"), idValue)
+                    assertEquals(child, childInParent)
+
+                    val linkingParent = child.linkingObjects("EmbeddedSimpleParent", "child")
+                            .first()
+                    assertNotNull(linkingParent)
+                    assertEquals(parent.getString("id"), linkingParent!!.getString("id"))
+                    assertEquals(parent.getObject("child"), linkingParent.getObject("child"))
+                }
+            }
+
+    @Test
+    fun dynamicRealm_createEmbeddedObject_simpleChildList() =
+            DynamicRealm.getInstance(realm.configuration).use { realm ->
+                realm.executeTransaction {
+                    val parent = realm.createObject("EmbeddedSimpleListParent", UUID.randomUUID().toString())
+                    val child1 = realm.createEmbeddedObject("EmbeddedSimpleChild", parent, "children")
+                    val child2 = realm.createEmbeddedObject("EmbeddedSimpleChild", parent, "children")
+                    assertEquals(2, parent.getList("children").size.toLong())
+                    assertEquals(child1, parent.getList("children").first()!!)
+                    assertEquals(child2, parent.getList("children").last()!!)
+                }
+            }
+
+    @Test
+    fun dynamicRealm_createEmbeddedObject_wrongParentPropertyTypeThrows() {
+        DynamicRealm.getInstance(realm.configuration).use { realm ->
+            realm.executeTransaction {
+                val parent = realm.createObject("EmbeddedSimpleParent", "parent")
+
+                // TODO: Smoke-test for wrong type. Figure out how to test all unsupported types.
```

Comment:
If you think this test is sufficient, just remove the TODO, otherwise, we need to implement a test for all the other "wrong" types.